### PR TITLE
feat(Tree): enable drag & drop ordering of tree nodes

### PR DESCRIPTION
**Pull request by Kenny806**

### CODE REVIEW

File: src/components/Tree/Node.tsx [hunk 1]
```diff
@@ -56,81 +91,98 @@ export const TreeNode = ({
...
-                            <TreeNode
-                                key={node.id}
-                                node={node}
-                                activeNodeId={activeNodeId}
-                                onClick={onClick}
-                                parentIds={[...parentIds, id]}
-                            />
-                        );
-                    })}
+                    {renderNodesArray(nodes, activeNodeId, onClick, onDrop, [...parentIds, id])}
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:176]: renderNodesArray maybe better name should be renderNodes or renderNodeArray. As this way its redundant to call it nodes with array.

File: src/components/Tree/Tree.spec.tsx [hunk 2]
```diff
@@ -10,10 +10,18 @@ import IconGuidelines from "@foundation/Icon/Generated/IconGuidelines";
...
 };
-const Component: FC<ComponentProps> = ({ nodes }) => {
+const Component: FC<ComponentProps> = ({ nodes, onDrop }) => {
     const [selectedId, setSelectedId] = useState<string | null>();
-    return <Tree nodes={nodes} activeNodeId={selectedId} onSelect={(id: string | null) => setSelectedId(id)} />;
+    return (
+        <Tree
+            nodes={nodes}
+            activeNodeId={selectedId}
+            onSelect={(id: string | null) => setSelectedId(id)}
```
Comment (GoranRibic) [inline src/components/Tree/Tree.spec.tsx:21]: use NullableString instead of string | null

File: src/components/Tree/Tree.stories.tsx [hunk 3]
```diff
@@ -5,208 +5,44 @@ import { Meta, Story } from "@storybook/react";
...
-                        ],
-                    },
-                ]}
-            />
-        </Flyout>
-    );
+interface TreeListItem extends TreeNodeProps {
+    name: string;
+    parentId: TreeNodeProps["id"] | null;
+    sort: number | null;
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:15]: NullableNumber

File: src/components/Tree/utils/ActionsFlyoutComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+import { Flyout } from "@components/Flyout";
+import { Button, ButtonSize, ButtonStyle } from "@components/Button";
+import { IconActions, IconSize } from "@foundation/Icon";
+import { ActionMenu } from "@components/ActionMenu/ActionMenu";
+import { mockActionMenuBlocks } from "@components/Tree/utils/mocks";
+
+export const ActionsFlyoutComponent = () => {
+    const [open, setOpen] = useState(false);
+    const onClick = () => {
```
Comment (GoranRibic) [inline src/components/Tree/utils/ActionsFlyoutComponent.tsx:10]: rename onClick to handleClick

Comment (Kenny806) [reply src/components/Tree/utils/ActionsFlyoutComponent.tsx:10]: Done

File: src/components/Tree/Tree.spec.tsx [hunk 5]
```diff
@@ -139,3 +148,36 @@ describe("Tree Component", () => {
         cy.get(`@NextSelectedItem`).should("have.attr", "aria-selected", "true");
     });
 });
+
+describe("Draggable Tree Component", () => {
+    it("renders correct drop zones on the top level", () => {
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/Tree/Tree.spec.tsx:153]: Added TODO comments (basic Tree behavior withour onDrop is already tested)

File: src/components/Tree/Tree.spec.tsx (hunk 2 above)
Comment (Kenny806) [reply src/components/Tree/Tree.spec.tsx:21]: Nullable* types are not in arcade ATM. That kind of change should have a separate commit (it's not related to changes done in this PR). You can check #641 , it's all done there.

File: src/components/Tree/Tree.stories.tsx (hunk 3 above)
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx:15]: Nullable* types are not in arcade ATM. That kind of change should have a separate commit (it's not related to changes done in this PR). You can check #641 , it's all done there.

File: src/components/Tree/Tree.stories.tsx [hunk 6]
```diff
@@ -5,208 +5,44 @@ import { Meta, Story } from "@storybook/react";
...
-                            />,
-                            <ActionsFlyoutComponent key="actions" />,
-                        ],
-                    },
-                ],
-            },
-        ],
-    },
-];
+const listToTree = (items: TreeListItem[], id: string | null = null): TreeNodeProps[] =>
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx:30]: Done, icons are now hardcoded in the mock data

**Review by GoranRibic — APPROVED** [review]

**Pull request merged**